internal/tasks/schedulertask.go

```diff
@@ -203,17 +203,27 @@ func (sm *SchedulerTask) refreshNodeBindings(ctx context.Context) error {
 		return err
 	}
 
-	// Delete any binding where subscription is removed, This needs to be handled before nodes updates
+	// Delete any binding where subscription is removed or
+	// the subscription's current version id doesn't match the node binding's subscription version id.
+	// This needs to be handled before nodes updates
 	// as node update will cause subscriptions to be rescheduled on other nodes
 	var validBindings []*nodebinding.Model
 
 	// TODO: Optimize O(N*M) to O(N+M)
 	for _, nb := range nodeBindings {
 		found := false
 		for _, sub := range sm.subCache {
+			sVID, verr := sub.GetVersionID()
+			if verr != nil {
+				return verr
+			}
 			if sub.Name == nb.SubscriptionID {
-				found = true
-				break
+				if nb.SubscriptionVersionID == sVID {
```

Comment:
merge with if above than nesting ?

Reply:
This was done on purpose so that in case where the subscription matches but the version does not match, we print a log saying the version changed. If I have to merge:
```
if (subscription.name = binding.subscriptionid) && (binding.subscriptionVersion == subscriptionVersion){
found = true
break
}
if (subscription.name = binding.subscriptionid) && (binding.subscriptionVersion != subscriptionVersion){
print log("Subscription matched but version did not match. Subscription has been updated.")
}
```
This I felt is less readable/difficult to understand